### PR TITLE
bug fixed and version updated

### PR DESCRIPTION
**Pull request by bugrabuga**

### CODE REVIEW

**Comment by bugrabuga** [comment]
Hi,

I don't understand what you mean by changes in formatting, from single quotes to double quotes?
Python 3.8 version does not work on GCP, so I updated Python to the latest version, most of the changes in libraries are related to deprecating the version, while creating the dataflow template, BigQueryWrapper in bq_mod now includes a priority parameter, so it was necessary to add this, we tested the changes and went over them with the Google Turkey team, can you review the changes again?

Thanks again !

**Comment by bugrabuga** [comment]
I changed the double quotes in the setup file to single quotes and rolled back the python version. Currently there are only version updates and parameter updates. Can you review it again?

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 1]
```diff
@@ -34,31 +34,31 @@
 from apache_beam.options.value_provider import ValueProvider
 from apache_beam.options.value_provider import check_accessible
 
-FieldSchema = collections.namedtuple('FieldSchema', 'fields mode name type')
+FieldSchema = collections.namedtuple("FieldSchema", "fields mode name type")
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:37]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 2]
```diff
@@ -34,31 +34,31 @@
...
 
 
 def _to_bool(value):
-    return value == 'true'
+    return value == "true"
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:41]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 3]
```diff
@@ -34,31 +34,31 @@
...
 
 
 def _to_decimal(value):
     return decimal.Decimal(value)
 
 
 def _to_bytes(value):
-    return value.encode('utf-8')
+    return value.encode("utf-8")
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:49]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 4]
```diff
@@ -34,31 +34,31 @@
...
-            'BOOLEAN': _to_bool,
-            'NUMERIC': _to_decimal,
-            'BYTES': _to_bytes,
+            "INTEGER": int,
+            "INT64": int,
+            "FLOAT": float,
+            "BOOLEAN": _to_bool,
+            "NUMERIC": _to_decimal,
+            "BYTES": _to_bytes,
         }
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:62]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 5]
```diff
@@ -72,7 +72,7 @@ def _convert_to_tuple(cls, table_field_schemas):
         ]
 
     def decode(self, value):
-        value = json.loads(value.decode('utf-8'))
+        value = json.loads(value.decode("utf-8"))
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:75]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 6]
```diff
@@ -81,9 +81,10 @@ def _decode_with_schema(self, value, schema_fields):
                 value[field.name] = None
                 continue
 
-            if field.type == 'RECORD':
+            if field.type == "RECORD":
                 value[field.name] = self._decode_with_schema(
-                    value[field.name], field.fields)
+                    value[field.name], field.fields
+                )
             else:
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:88]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 7]
```diff
@@ -101,27 +102,31 @@ def to_type_hint(self):
...
             raise ValueError(
-                'Both a BigQuery table and a query were specified.'
-                ' Please specify only one of these.')
+                "Both a BigQuery table and a query were specified."
+                " Please specify only one of these."
+            )
         elif table is None and query is None:
-            raise ValueError('A BigQuery table or a query must be specified')
+            raise ValueError("A BigQuery table or a query must be specified")
         elif table is not None:
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:126]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 8]
```diff
@@ -101,27 +102,31 @@ def to_type_hint(self):
...
             self.table_reference = bigquery_tools.parse_table_reference(
-                table, dataset, project)
+                table, dataset, project
+            )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:129]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 9]
```diff
@@ -143,14 +149,15 @@ def __init__(
         self.kms_key = kms_key
         self.split_result = None
 
-    @check_accessible(['query'])
+    @check_accessible(["query"])
     def estimate_size(self):
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:153]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 10]
```diff
@@ -143,14 +149,15 @@ def __init__(
...
         bq = bigquery_tools.BigQueryWrapper()
         if self.table_reference is not None:
             table = bq.get_table(
                 self.table_reference.projectId,
                 self.table_reference.datasetId,
-                self.table_reference.tableId)
+                self.table_reference.tableId,
+            )
             return int(table.numBytes)
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:161]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 11]
```diff
@@ -179,7 +188,9 @@ def split(self, desired_bundle_size, start_position=None, stop_position=None):
                     0,
                     CompressionTypes.UNCOMPRESSED,
                     True,
-                    self.coder(schema)) for metadata in metadata_list
+                    self.coder(schema),
+                )
+                for metadata in metadata_list
             ]
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:194]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 12]
```diff
@@ -200,23 +211,26 @@ def stop_position(self):
         return CustomBigQuerySourceRangeTracker()
 
     def read(self, range_tracker):
-        raise NotImplementedError('BigQuery source must be split before being read')
+        raise NotImplementedError("BigQuery source must be split before being read")
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:215]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 13]
```diff
@@ -200,23 +211,26 @@ def stop_position(self):
...
 
-    @check_accessible(['query'])
+    @check_accessible(["query"])
     def _setup_temporary_dataset(self, bq):
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:217]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 14]
```diff
@@ -200,23 +211,26 @@ def stop_position(self):
...
         location = bq.get_query_location(
-            self.project.get(), self.query.get(), self.use_legacy_sql)
+            self.project.get(), self.query.get(), self.use_legacy_sql
+        )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:220]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 15]
```diff
@@ -200,23 +211,26 @@ def stop_position(self):
...
         bq.create_temporary_dataset(self.project.get(), location)
 
-    @check_accessible(['query'])
+    @check_accessible(["query"])
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:223]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 16]
```diff
@@ -229,18 +243,21 @@ def _export_files(self, bq):
...
-                                         bigquery_tools.FileFormat.JSON,
-                                         include_header=False)
+        job_ref = bq.perform_extract_job(
+            [gcs_location],
+            job_id,
+            self.table_reference,
+            bigquery_tools.FileFormat.JSON,
+            include_header=False,
+        )
         bq.wait_for_bq_job(job_ref)
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:253]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 17]
```diff
@@ -229,18 +243,21 @@ def _export_files(self, bq):
...
         metadata_list = FileSystems.match([gcs_location])[0].metadata_list
 
         table = bq.get_table(
             self.table_reference.projectId,
             self.table_reference.datasetId,
-            self.table_reference.tableId)
+            self.table_reference.tableId,
+        )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:260]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 18]
```diff
@@ -255,15 +272,16 @@ def process(self, element):
                 yield element
 
         output = input | beam.ParDo(PassThrough()).with_outputs(
-            'cleanup_signal', main='main')
-        main_output = output['main']
-        cleanup_signal = output['cleanup_signal']
+            "cleanup_signal", main="main"
+        )
+        main_output = output["main"]
+        cleanup_signal = output["cleanup_signal"]
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:278]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 19]
```diff
@@ -255,15 +272,16 @@ def process(self, element):
...
 
         _ = (
             input.pipeline
             | beam.Create([None])
-            | beam.ParDo(
-                self.cleanup_dofn, beam.pvalue.AsSingleton(cleanup_signal)))
+            | beam.ParDo(self.cleanup_dofn, beam.pvalue.AsSingleton(cleanup_signal))
+        )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:284]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 20]
```diff
@@ -273,9 +291,10 @@ def __init__(self, gcs_location=None, validate=False, *args, **kwargs):
         if gcs_location:
             if not isinstance(gcs_location, (str, unicode, ValueProvider)):
                 raise TypeError(
-                    '%s: gcs_location must be of type string'
-                    ' or ValueProvider; got %r instead' %
-                    (self.__class__.__name__, type(gcs_location)))
+                    "%s: gcs_location must be of type string"
+                    " or ValueProvider; got %r instead"
+                    % (self.__class__.__name__, type(gcs_location))
+                )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:297]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 21]
```diff
@@ -291,25 +310,27 @@ def _get_destination_uri(self):
         """Returns the fully qualified Google Cloud Storage URI where the
         extracted table should be written.
         """
-        file_pattern = 'bigquery-table-dump-*.json'
-        gcs_base = ''
+        file_pattern = "bigquery-table-dump-*.json"
+        gcs_base = ""
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:314]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 22]
```diff
@@ -291,25 +310,27 @@ def _get_destination_uri(self):
...
             if self.gcs_location.is_accessible():
                 gcs_base = self.gcs_location.get()
         else:
             raise ValueError(
-                '{} requires a GCS location to be provided'.format(
-                    self.__class__.__name__))
+                "{} requires a GCS location to be provided".format(
+                    self.__class__.__name__
+                )
+            )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:324]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 23]
```diff
@@ -291,25 +310,27 @@ def _get_destination_uri(self):
...
             self._validate_gcs_location(gcs_base)
 
         return FileSystems.join(gcs_base, self.job_id, file_pattern)
 
     @staticmethod
     def _validate_gcs_location(gcs_location):
-        if not gcs_location.startswith('gs://'):
-            raise ValueError('Invalid GCS location: {}'.format(gcs_location))
+        if not gcs_location.startswith("gs://"):
+            raise ValueError("Invalid GCS location: {}".format(gcs_location))
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:333]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 24]
```diff
@@ -320,7 +341,8 @@ def process(self, unused_element, signal):
                 gcs_location = self.get_destination_uri()
                 match_result = FileSystems.match([gcs_location])[0].metadata_list
                 logging.debug(
-                    "%s: matched %s files", self.__class__.__name__, len(match_result))
+                    "%s: matched %s files", self.__class__.__name__, len(match_result)
+                )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:345]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py [hunk 25]
```diff
@@ -332,5 +354,8 @@ def process(self, unused_element, signal):
                     get_destination_uri=self._get_destination_uri,
                     validate=self.validate,
                     *self._args,
-                    **self._kwargs))
-            | _PassThroughThenCleanup(RemoveJsonFiles(self._get_destination_uri)))
+                    **self._kwargs
+                )
+            )
+            | _PassThroughThenCleanup(RemoveJsonFiles(self._get_destination_uri))
+        )
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/bigquery_mod/__init__.py:361]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/install.sh [hunk 26]
```diff
@@ -175,4 +175,4 @@ install_dependencies
 create_gcs_bucket
 install_dataflow_templates
 
-printf "\nEND - FoCVS installation completed successfully!\n"
+printf "\nEND - FoCVS installation completed successfully!\n"
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/install.sh:178]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/requirements.txt [hunk 27]
```diff
@@ -1,829 +1,2099 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.12
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/requirements.txt:2]: regenerate using Python 3.8

File: marketing-analytics/predicting/future-customer-value-segments/setup.py [hunk 28]
```diff
@@ -15,11 +15,11 @@
...
-    name='Future-Customer-Value-Segments',
-    version='1.0.0',
+    name="Future-Customer-Value-Segments",
+    version="1.0.0",
     install_requires=[
-        'Lifetimes==0.11.3',
-        'matplotlib==3.3.4',
+        "Lifetimes==0.11.3",
+        "matplotlib==3.3.4",
     ],
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/setup.py:23]: please revert

File: marketing-analytics/predicting/future-customer-value-segments/setup.py [hunk 29]
```diff
@@ -15,11 +15,11 @@
 import setuptools
 
 setuptools.setup(
-    name='Future-Customer-Value-Segments',
-    version='1.0.0',
+    name="Future-Customer-Value-Segments",
+    version="1.0.0",
```
Comment (mohabfekry) [inline marketing-analytics/predicting/future-customer-value-segments/setup.py:19]: please revert

**Review by mohabfekry — CHANGES_REQUESTED** [review]

**Review by mohabfekry — APPROVED** [review]
Thanks for the update!

**Comment by bugrabuga** [comment]
This is my first contribution to an open-source project, please forgive my question, but do I need to do anything else from now on or will my contribution be published in this project :)

**Comment by mohabfekry** [comment]
No that's all, your contribution will be merged very soon as we're running some final internal checks. Thank you!

**Comment by bugrabuga** [comment]
Thank you !

**Pull request merged**